app/src/main/java/com/android/voyageur/ui/search/SearchUserProfile.kt

```diff
@@ -0,0 +1,175 @@
+ * @param navigationActions The NavigationActions used for handling navigation actions within the
+ *   app.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchUserProfileScreen(userViewModel: UserViewModel, navigationActions: NavigationActions) {
+  // Observing the selected user and loading state from the ViewModel
+  val user by userViewModel.selectedUser.collectAsState()
+  val isLoading by userViewModel.isLoading.collectAsState()
+
+  // Navigate back to search if no user data is available and loading is complete
+  if (user == null && !isLoading) {
+    LaunchedEffect(Unit) { navigationActions.navigateTo(Route.SEARCH) }
+    return
+  }
+
+  // Scaffold layout containing top bar for navigation and content for displaying user details
+  Scaffold(
+      modifier = Modifier.testTag("userProfileScreen"),
+      topBar = {
+        TopAppBar(
+            title = { Text("User Profile") },
+            navigationIcon = {
+              IconButton(
+                  onClick = {
+                    userViewModel.deselectUser() // Clears selected user in the ViewModel
+                    navigationActions.goBack() // Navigate back
+                  }) {
+                    Icon(
+                        imageVector = Icons.Default.ArrowBack,
+                        contentDescription = "Back",
+                        modifier = Modifier.testTag("userProfileBackButton"))
+                  }
+            })
+      },
+      content = { paddingValues ->
+        Box(
+            modifier =
+                Modifier.fillMaxSize().padding(paddingValues).testTag("userProfileScreenContent"),
+            contentAlignment = Alignment.Center) {
+              // Display appropriate content based on loading and user state
+              when {
+                isLoading -> {
+                  CircularProgressIndicator(
+                      modifier = Modifier.testTag("userProfileLoadingIndicator"))
+                }
+                user != null -> {
+                  SearchUserProfileContent(userData = user!!, userViewModel = userViewModel)
+                }
+                else -> {
+                  Text("No user data available", modifier = Modifier.testTag("userProfileNoData"))
+                }
+              }
+            }
+      })
+}
+
+/**
+ * Composable function for displaying the details of a selected user. Displays profile picture,
+ * name, email, interests, and an "Add Contact" button.
+ *
+ * @param userData The selected User object containing user details.
+ * @param userViewModel The UserViewModel used for managing user data and interactions.
+ */
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun SearchUserProfileContent(userData: User, userViewModel: UserViewModel) {
+  // Observing if the selected user is already a contact
+  val isContactAdded by remember {
+    derivedStateOf { userViewModel.user.value?.contacts?.contains(userData.id) ?: false }
+  }
+
+  Column(
```

Comment:
Here, almost all the code is duplicated from the ProfileContent composable in ProfileScreen.kt. I think you could use the same composable from ProfileScreen and change the parameters.

Reply:
I resolved this by creating a new composable component that can be used on both screens. Thank you very much for the suggestion!